sdk/servicebus/azure-messaging-servicebus/src/samples/java/com/azure/messaging/servicebus/ReadmeSamples.java

```diff
@@ -55,19 +59,118 @@ public void createAsynchronousServiceBusReceiverWithAzureIdentity() {
+            ServiceBusReceivedMessage message = context.getMessage();
+            System.out.printf("Id: %s. Contents: %s%n", message.getMessageId(),
+                new String(message.getBody(), StandardCharsets.UTF_8));
+        });
+    }
+
+    /**
+     * Receives messages asynchronously.
+     */
+    public void receiveMessagesAsync() {
         ServiceBusReceiverAsyncClient receiver = new ServiceBusClientBuilder()
-            .credential("<<fully-qualified-namespace>>", credential)
+            .connectionString("<< CONNECTION STRING FOR THE SERVICE BUS NAMESPACE >>")
             .receiver()
-            .queueName("<<queue-name>>")
+            .queueName("<< QUEUE NAME >>")
+            .buildAsyncClient();
+
+        Disposable subscription = receiver.receive().subscribe(context -> {
+            ServiceBusReceivedMessage message = context.getMessage();
+            System.out.printf("Id: %s%n", message.getMessageId());
+            System.out.printf("Contents: %s%n", new String(message.getBody(), StandardCharsets.UTF_8));
+        }, error -> {
+                System.err.println("Error occurred while receiving messages: " + error);
+            }, () -> {
+                System.out.println("Finished receiving messages.");
+            });
+    }
+
+    /**
+     * Complete a message.
+     */
+    public void completeMessage() {
+        ServiceBusReceiverClient receiver = new ServiceBusClientBuilder()
+            .connectionString("<< CONNECTION STRING FOR THE SERVICE BUS NAMESPACE >>")
+            .receiver()
+            .topicName("<< TOPIC NAME >>")
+            .subscriptionName("<< SUBSCRIPTION NAME >>")
+            .receiveMode(ReceiveMode.PEEK_LOCK)
+            .buildClient();
+
+        receiver.receive(10).forEach(context -> {
+            ServiceBusReceivedMessage message = context.getMessage();
+
+            // Process message and then complete it.
+            receiver.complete(message);
+        });
+    }
+
+    /**
+     * Create a session message.
+     */
+    public void createSessionMessage() {
+        ServiceBusSenderClient sender = new ServiceBusClientBuilder()
+            .connectionString("<< CONNECTION STRING FOR THE SERVICE BUS NAMESPACE >>")
+            .sender()
+            .queueName("<< QUEUE NAME >>")
+            .buildClient();
+
+        ServiceBusMessage message = new ServiceBusMessage("Hello world".getBytes())
+            .setSessionId("greetings");
+
+        sender.send(message);
+    }
+
+    /**
+     * Create session receiver for "greetings"
+     */
+    public void namedSessionReceiver() {
+        ServiceBusReceiverAsyncClient receiver = new ServiceBusClientBuilder()
+            .connectionString("<< CONNECTION STRING FOR THE SERVICE BUS NAMESPACE >>")
+            .sessionReceiver()
+            .queueName("<< QUEUE NAME >>")
+            .sessionId("greetings")
+            .buildAsyncClient();
+    }
+
+    /**
+     * Create session receiver for "greetings"
```

Comment:
_might_ be true? :) I mean, it's unnamed so it's whatever the next available session is. 

Not sure if this is a copy/paste error or not but it might warrant some explanatory text either way.